src/sdk/pynni/nni/nas/pytorch/base_mutator.py

```diff
@@ -0,0 +1,127 @@
+import logging
+
+import torch.nn as nn
+from nni.nas.pytorch.mutables import Mutable, MutableScope, InputChoice
+from nni.nas.pytorch.utils import StructuredMutableTreeNode
+
+logger = logging.getLogger(__name__)
+
+
+class BaseMutator(nn.Module):
+    """
+    A mutator is responsible for mutating a graph by obtaining the search space from the network and implementing
+    callbacks that are called in ``forward`` in Mutables.
+    """
+
+    def __init__(self, model):
+        super().__init__()
+        self.__dict__["model"] = model
+        self._structured_mutables = self._parse_search_space(self.model)
+
+    def _parse_search_space(self, module, root=None, prefix="", memo=None, nested_detection=None):
+        if memo is None:
```

Comment:
if not memo?

Reply:
memo is `{}` is different from memo is None. memo is a state dict that catches already visited nodes.